autoscale-scheduler/cmd/main.go

```diff
@@ -65,7 +67,9 @@ func runProgram(logger *zap.Logger) (err error) {
 	// everything fit nicely, we'll redirect it to zap as well.
 	redirectKlog(logger.Named("klog"))
 
-	constructor := plugin.NewAutoscaleEnforcerPlugin(ctx, logger, conf)
+	constructor := func(ctx context.Context, configuration runtime.Object, f framework.Handle) (framework.Plugin, error) {
+		return plugin.NewAutoscaleEnforcerPlugin(ctx, logger, conf)(configuration, f)
+	}
 	command := app.NewSchedulerCommand(app.WithPlugin(plugin.Name, constructor))
```

Comment:
So, apparently the function signature changed somewhere with the upgrade - such that if I'm understanding correctly, `withPlugin` is expected to take a constructor that takes a `ctx`, not just `configuration` and `framework.Handle`. Not sure what the right fix for that is, so I'm trying wrapping it. This is also the solution copilot suggested, but it's probable we're both dumb 😆

Reply:
Yeah! Looks like you're right! And since it's only used in this one spot it was easy to clean up (and even remove the unused `_obj` parameter)